### ISSUE TITLE
[opam] upper bound on cohttp

### ISSUE THREAD

**Issue by jrochel** [inhabitedtype/ocaml-aws#96]
In the opam repository of this package (https://opam.ocaml.org/packages/aws/) there is an upper version limit (implemented as a conflict) on the cohttp package: 

```
conflicts: [
  "cohttp" {< "0.17.0"}
  "cohttp" {>="0.99"}
]
```

Is there any good reason for this? Everything compiles just fine for me with `cohttp.2.5.4`.

**Comment by kit-ty-kate** [comment]
Added details from https://github.com/ocaml/opam-repository/issues/18212#issuecomment-784333821: the latest available version does not compile with cohttp + lwt + ssl so the constraints are correct.

I would encourage the developers here to make a new release as the master branch seems to be alright.

**Comment by tmcgilchrist** [comment]
Thanks for the report @jrochel I'm working on putting a new version onto opam. 
This issue https://github.com/inhabitedtype/ocaml-aws/pull/83 tracks the outstanding tasks.

Which AWS services are you interested in using?

**Comment by jrochel** [comment]
We are currently using a whole bunch: S3, cloudformation, sns, sqs, rds, cloudwatch, Elasticsearch, Lambda, Ses, Mediaconvert, ssm. But we are using our own library, which is hand-coded and only very partial, and ugly, too ugly for publishing.

I'm currently working on adding DynamoDB to the mix and I would very much like to use your package `aws` for it. If everything works out fine, I'll probably work on porting our `bs-aws` package to `aws`, this adding some functorial interfaces and abstractions on top.

I'll probably make a PR soon, however I'm having some troubles with mutual recursion in the botocore specifications. I'll work a bit on a solution, which I'd like to discuss with you guys.

**Comment by tmcgilchrist** [comment]
This should be resolved in the latest 1.2 release on OPAM https://opam.ocaml.org/packages/aws/
Please reopen if the problem still exists.
